ext/MathOptAIIpoptPythonCallExt.jl

```diff
@@ -0,0 +1,145 @@
+# Copyright (c) 2024: Triad National Security, LLC
+# Copyright (c) 2024: Oscar Dowson and contributors
+#
+# Use of this source code is governed by a BSD-style license that can be found
+# in the LICENSE.md file.
+
+module MathOptAIIpoptPythonCallExt
+
+import JuMP
+import Ipopt
+import PythonCall
+import MathOptAI
+
+"""
+    MathOptAI.add_predictor(
+        model::JuMP.AbstractModel,
+        predictor::MathOptAI.VectorNonlinearOracle{MathOptAI.PytorchModel},
+        x::Vector;
+        device::String = "cpu",
+    )
+
+Add a trained neural network from PyTorch via PythonCall.jl to `model`.
+
+## Keyword arguments
+
+ * `device`: device used to construct PyTorch tensors, e.g. `"cuda"` to run on
+   an Nvidia GPU.
+"""
+function MathOptAI.add_predictor(
+    model::JuMP.AbstractModel,
+    predictor::MathOptAI.VectorNonlinearOracle{MathOptAI.PytorchModel},
+    x::Vector;
+    device::String = "cpu",
+)
+    set = _build_set_and_callback(predictor.predictor, length(x); device)
+    y = JuMP.@variable(model, [1:set.output_dimension])
+    con = JuMP.@constraint(model, [x; y] in set)
+    return y, MathOptAI.Formulation(predictor, y, [con])
+end
+
+function _build_set_and_callback(
+    predictor::MathOptAI.PytorchModel,
+    input_dimension::Int;
+    device::String,
+)
+    torch = PythonCall.pyimport("torch")
+    torch_model = torch.load(predictor.filename; weights_only = false)
+    torch_model = torch_model.to(device)
+    J = torch.func.jacrev(torch_model)
+    y = torch_model(torch.zeros(input_dimension; device))
```

Comment:
This syntax uses the local name of the variable, "device", as the keyword in the call to the Python function?

Reply:
Yip. Just avoids `device = device`
```julia
julia> foo(; x) = x
foo (generic function with 1 method)

julia> x = 1
1

julia> foo(; x = x)
1

julia> foo(; x)
1
```